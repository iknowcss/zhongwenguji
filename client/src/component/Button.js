import React from 'react';
import cx from 'classnames';
import style from './Button.module.scss';

export default ({ secondary, accent, ...props }) => (
  <button {...props} className={cx(style.button, {
    [style.secondary]: secondary,
    [style.accent]: accent,
  }, props.className)} />
);
