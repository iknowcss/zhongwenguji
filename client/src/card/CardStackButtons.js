import React, { Component } from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import Button from '../component/Button';
import { SearchIcon, UndoIcon } from '../component/Icon';
import style from './CardStackButtons.module.scss';
import noop from '../util/noop';

export default class CardStackButtons extends Component {
  static propTypes = {
    onUndo: PropTypes.func,
    onDefinition: PropTypes.func,
    showDefinition: PropTypes.bool
  };

  static defaultProps = {
    onUndo: noop,
    onDefinition: noop,
    showDefinition: false
  };

  handleUndoClick = (event) => {
    event.preventDefault();
    this.props.onUndo();
  };

  handleDefinitionClick = (event) => {
    event.preventDefault();
    this.props.onDefinition();
  };

  render() {
    return (
      <div className={style.buttonContainer}>
        <Button
          disabled={this.props.showDefinition}
          className={cx(style.button, style.undoButton)}
          onClick={this.handleUndoClick}
        >
          <UndoIcon size="medium" className={style.buttonIcon} />
        </Button>
        <Button
          secondary={this.props.showDefinition}
          className={cx(style.button)}
          onClick={this.handleDefinitionClick}
        >
          <SearchIcon size="medium" className={style.buttonIcon} />
        </Button>
      </div>
    );
  }
}
