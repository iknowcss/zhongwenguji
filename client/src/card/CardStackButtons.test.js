import React from 'react';
import Renderer from 'react-test-renderer';
import Button from '../component/Button';
import CardStackButtons from './CardStackButtons';

describe('CardStackButtons', () => {
  let component;
  let undoButton;
  let definitionButton;

  function setup(props) {
    component = Renderer.create(<CardStackButtons {...props} />);
    const buttons = component.root.findAllByType(Button);
    undoButton = buttons[0];
    definitionButton = buttons[1];
  }

  it('renders', () => {
    setup();
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('updates the buttons when showDefinition is true', () => {
    setup({ showDefinition: true });
    expect(undoButton.props.disabled).toEqual(true);
    expect(definitionButton.props.className.split(/\s+/)).toEqual(
      expect.arrayContaining(['buttonActive'])
    );
  });

  describe('click handlers', () => {
    let onUndo;
    let onDefinition;
    let preventDefault;

    beforeEach(() => {
      onUndo = jest.fn();
      onDefinition = jest.fn();
      preventDefault = jest.fn();
      setup({ onUndo, onDefinition });
    });

    it('calls onUndo', () => {
      undoButton.props.onClick({ preventDefault });
      expect(preventDefault).toHaveBeenCalledTimes(1);
      expect(onUndo).toHaveBeenCalledTimes(1);
    });

    it('calls onDefinition', () => {
      definitionButton.props.onClick({ preventDefault });
      expect(preventDefault).toHaveBeenCalledTimes(1);
      expect(onDefinition).toHaveBeenCalledTimes(1);
    });
  });
});
