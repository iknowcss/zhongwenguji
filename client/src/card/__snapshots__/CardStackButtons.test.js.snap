// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardStackButtons renders 1`] = `
<div
  className="buttonContainer"
>
  <button
    className="button button undoButton"
    onClick={[Function]}
  >
    <span
      className="span span--medium undoIcon buttonIcon"
    >
      <svg
        className="svg"
        viewBox="0 0 511.63 511.631"
      >
        <use
          xlinkHref="undo.svg#main"
        />
      </svg>
    </span>
  </button>
  <button
    className="button button"
    onClick={[Function]}
  >
    <span
      className="span span--medium searchIcon buttonIcon"
    >
      <svg
        className="svg"
        viewBox="0 0 250.313 250.313"
      >
        <use
          xlinkHref="search.svg#main"
        />
      </svg>
    </span>
  </button>
</div>
`;
